contracts/test/PDPService.t.sol

```diff
@@ -313,6 +316,133 @@ contract PDPServiceProofSetMutateTest is Test {
     }
 }
 
+contract PDPServiceProofTest is Test {
+    uint256 constant challengeFinalityDelay = 2;
+    string constant cidPrefix = "CID";
+    PDPService pdpService;
+
+    function setUp() public {
+        pdpService = new PDPService(challengeFinalityDelay);
+    }
+
+    function testProveSingleRoot() public {
+        uint leafCount = 10;
+        bytes32[][] memory tree = makeTree(leafCount);
+        // console.log("root          ", vm.toString(tree[0][0]));
+
+        // Create new proof set and add root.
+        uint256 setId = pdpService.createProofSet();
+        PDPService.RootData[] memory roots = new PDPService.RootData[](1);
+        roots[0] = makeRoot(tree, leafCount);
+        pdpService.addRoots(setId, roots);
+
+        // Advance chain until challenge epoch.
+        uint256 challengeEpoch = pdpService.getNextChallengeEpoch(setId);
+        vm.roll(challengeEpoch);
+
+        // Build a proof with  multiple challenges to single tree.
+        uint challengeCount = 3;
+        bytes32[][][] memory trees = new bytes32[][][](1);
+        trees[0] = tree;
+        uint[] memory leafCounts = new uint[](1);
+        leafCounts[0] = leafCount;
+        PDPService.Proof[] memory proofs = buildProofs(setId, challengeCount, trees, leafCounts);
+
+        // Submit proof.
+        pdpService.provePossession(setId, proofs);
+        // Verify the next challenge is in a subsequent epoch.
+        assertEq(pdpService.getNextChallengeEpoch(setId), block.number + challengeFinalityDelay);
+    }
+
+    function testProveManyRoots() public {
+        // Build Merkle trees.
+        uint rootCount = 3;
+        uint[] memory leafCounts = new uint[](rootCount);
+        bytes32[][][] memory trees = new bytes32[][][](rootCount);
+        PDPService.RootData[] memory roots = new PDPService.RootData[](rootCount);
+        for (uint i = 0; i < rootCount; i++) {
+            // Generate different size trees up to max leaves.
```

Comment:
The protocol constant for max leaves is 1 << 50 so this comment is confusing

Reply:
I don't want to build a 2^50 merkle tree in a test. I'll clarify this inline constant.